cfme/middleware/jdr_report.py

```diff
@@ -0,0 +1,141 @@
+import re
+
+from datetime import datetime
+
+from cfme.utils import attributize_string
+from cfme.utils.appliance import current_appliance
+
+
+def _db_select_query(server):
+    """column order: `id`, `queued_at`, `requesting_user`, `status`"""
+    t_ms = current_appliance.db.client['middleware_servers']
+    t_mdr = current_appliance.db.client['middleware_diagnostic_reports']
+    query = current_appliance.db.client.session.query(
+        t_mdr.id, t_mdr.queued_at,
+        t_mdr.requesting_user.label('username'),
+        t_mdr.status)\
+        .join(t_mdr, t_mdr.middleware_server_id == t_ms.id)
+    if server:
+        query = query.filter(t_ms.name == server.name)
+    return query
+
+
+class JDRReport():
```

Comment:
**REQUIRED**
Please implement collections http://cfme-tests.readthedocs.io/guides/collections.html

Reply:
JDRReport itself is not an navigable object, it's tightly coupled to server and is not clickable listed rows for server.

Reply:
**REQUIRED**: It is not about navigation but about consistent API and dependency passing.